File: Module 7/73_Greedy/Python/CoinChange.py
def coin_change(deno, amt):
    nums = [0] * len(deno)
    for i in range(len(deno) - 1, -1, -1):
        while amt >= deno[i]:
            amt -= deno[i]
            nums[i] += 1

    print("The Denominations are:")
    for i in range(len(deno) - 1, -1, -1):
        if nums[i] > 0:
            print(f"{nums[i]} x {deno[i]}")

if __name__ == "__main__":
    deno = [1, 2, 5, 10, 20, 50, 100, 200, 500]
    amt = 2898

    coin_change(deno, amt)

File: Module 7/73_Greedy/Java/CoinChange.java
public class CoinChange {
    public static void coinChange(int[] deno, int amt) {
        int[] nums = new int[deno.length];
        for (int i = deno.length - 1; i >= 0; i--) {
            while (amt >= deno[i]) {
                amt -= deno[i];
                nums[i]++;
            }
        }
        System.out.println("The Denominations are:");
        for (int i = deno.length - 1; i >= 0; i--) {
            if (nums[i] > 0) {
                System.out.println(nums[i] + " x " + deno[i]);
            }
        }
    }

    public static void main(String[] args) {
        int[] deno = {1, 2, 5, 10, 20, 50, 100, 200, 500};
        int amt = 2898;

        coinChange(deno, amt);
    }
}
